packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -73,142 +77,168 @@ class LocationScreen extends HookWidget {
       [forecastsStateNotifier],
     );
 
-    return Scaffold(
-      resizeToAvoidBottomInset: false,
-      body: FloatingSearchAppBar(
-        automaticallyImplyBackButton: false,
-        controller: controller,
-        progress: isLoading.value,
-        onSubmitted: (String newCityName) async {
-          controller.close();
+    return LayoutBuilder(
+      builder: (context, constraints) => Scaffold(
+        resizeToAvoidBottomInset: false,
+        body: FloatingSearchAppBar(
+          automaticallyImplyBackButton: false,
+          controller: controller,
+          progress: isLoading.value,
+          onSubmitted: (String newCityName) async {
+            controller.close();
 
-          final trimmedCityName = newCityName.trim();
-          if (trimmedCityName.isEmpty) {
-            return;
-          }
+            final trimmedCityName = newCityName.trim();
+            if (trimmedCityName.isEmpty) {
+              return;
+            }
 
-          isLoading.value = true;
-          await cityStateNotifier.setCity(City(name: trimmedCityName));
-          if (context.read(c.cityStateNotifierProvider) is! c.Error) {
-            await Future.wait([
-              weatherStateNotifier.loadWeather(),
-              forecastsStateNotifier.loadForecasts(),
-            ]);
-          }
-          isLoading.value = false;
-        },
-        title: Text(
-          DateFormat('EEEE, d MMMM yyyy').format(DateTime.now()),
-          style: TextStyle(
-            color: Theme.of(context).textTheme.subtitle2.color,
-            fontSize: 14,
-          ),
-        ),
-        hint: 'Enter city name',
-        color: Theme.of(context).appBarTheme.color,
-        transitionCurve: Curves.easeInOut,
-        leadingActions: [
-          FloatingSearchBarAction(
-            child: CircularButton(
-              icon: Icon(
-                Icons.refresh,
-                color: Theme.of(context).appBarTheme.iconTheme.color,
-              ),
-              tooltip: 'Refresh',
-              onPressed: load,
+            isLoading.value = true;
+            await cityStateNotifier.setCity(City(name: trimmedCityName));
+            if (context.read(c.cityStateNotifierProvider) is! c.Error) {
+              await Future.wait([
+                weatherStateNotifier.loadWeather(),
+                forecastsStateNotifier.loadForecasts(),
+              ]);
+            }
+            isLoading.value = false;
+          },
+          title: Text(
+            DateFormat('EEEE, d MMMM yyyy').format(DateTime.now()),
+            style: TextStyle(
+              color: Theme.of(context).textTheme.subtitle2.color,
+              fontSize: constraints.maxWidth < kTabletBreakpoint ? 11.sp : 8.sp,
```

Comment:
Shouldn't this be `smallestWidth`? (https://stackoverflow.com/questions/49484549/can-we-check-the-device-to-be-smartphone-or-tablet-in-flutter)

Reply:
You are right.